src/afe.rs

```diff
@@ -0,0 +1,71 @@
+use embedded_hal;
+
+#[derive(Copy, Clone, Debug)]
+pub enum Gain {
+    G1 = 0b00,
+    G2 = 0b01,
+    G5 = 0b10,
+    G10 = 0b11
+}
+
+pub struct ProgrammableGainAmplifier<A0, A1> {
+    a0: A0,
+    a1: A1
+}
+
+impl<A0, A1> ProgrammableGainAmplifier<A0, A1>
+where
+    A0: embedded_hal::digital::v2::StatefulOutputPin,
+    A0::Error: core::fmt::Debug,
+    A1: embedded_hal::digital::v2::StatefulOutputPin,
+    A1::Error: core::fmt::Debug,
+{
+    pub fn new(a0: A0, a1: A1) -> Self
+    {
+        let mut afe = Self { a0: a0, a1: a1};
+
+        afe.set_gain(Gain::G1);
+
+        afe
+    }
+
+    pub fn set_gain(&mut self, gain: Gain) {
+        match gain {
```

Comment:
Since the Enum values already have the right bits, maybe this can be done without the match and with less code.
Same for `get_gain` below.

Reply:
This can be done for `set_gain()`, but it would require the `TryFrom` trait for `get_gain()`, which is (unfortunately) a `std` component. This could definitely be cleaner, but is generally restricted from being so based on the API provided by the `embedded-hal`. Namely, we can't do something like `set_state(true|false)` and instead have to use `set_high()` and `set_low()`.